packages/worker/src/sdk/users/users.ts

```diff
@@ -203,6 +228,131 @@ export const save = async (
   }
 }
 
+export const addTenant = async (
+  tenantId: string,
+  _id: string,
+  email: string
+) => {
+  if (env.MULTI_TENANCY) {
+    const afterCreateTenant = () =>
+      migrations.backPopulateMigrations({
+        type: MigrationType.GLOBAL,
+        tenantId,
+      })
+    await tenancy.tryAddTenant(tenantId, _id, email, afterCreateTenant)
+  }
+}
+
+export const bulkCreate = async (newUsersRequested: any[], groups: any) => {
+  const db = tenancy.getGlobalDB()
+  const tenantId = tenancy.getTenantId()
+
+  let usersToSave: any[] = []
+  let newUsers: any[] = []
+
+  const allUsers = await db.allDocs(
+    dbUtils.getGlobalUserParams(null, {
+      include_docs: true,
+    })
+  )
+  let mapped = allUsers.rows.map((row: any) => row.id)
+
+  const currentUserEmails = mapped.map((x: any) => x.email) || []
+  for (const newUser of newUsersRequested) {
+    if (
+      newUsers.find((x: any) => x.email === newUser.email) ||
+      currentUserEmails.includes(newUser.email)
+    ) {
+      continue
+    }
+    newUser.userGroups = groups
+    newUsers.push(newUser)
+  }
+
+  // Figure out how many builders we are adding and create the promises
+  // array that will be called by bulkDocs
+  let builderCount = 0
+  newUsers.forEach((user: any) => {
+    if (eventHelpers.isAddingBuilder(user, null)) {
+      builderCount++
+    }
+    usersToSave.push(
+      buildUser(
+        user,
+        {
+          hashPassword: true,
+          requirePassword: user.requirePassword,
+          bulkCreate: false,
+        },
+        tenantId
+      )
+    )
+  })
+
+  const usersToBulkSave = await Promise.all(usersToSave)
+  await quotas.addDevelopers(() => db.bulkDocs(usersToBulkSave), builderCount)
+
+  // Post processing of bulk added users, i.e events and cache operations
+  for (const user of usersToBulkSave) {
+    delete user.password
+    await eventHelpers.handleSaveEvents(user, null)
+    await apps.syncUserInApps(user._id)
+    await addTenant(tenantId, user._id, user.email)
```

Comment:
It's probably ok to leave this out, you shouldn't be able to bulk create without having first created the admin user. We should probably do that check like we do with the initial user creation to make sure. Just would be unnecessary db work to do the tenant creation logic for every user